src/ChunkMap.cpp

```diff
@@ -585,6 +595,11 @@ void cChunkMap::SetBlock(Vector3i a_BlockPos, BLOCKTYPE a_BlockType, NIBBLETYPE
 
 bool cChunkMap::GetBlockTypeMeta(Vector3i a_BlockPos, BLOCKTYPE & a_BlockType, NIBBLETYPE & a_BlockMeta) const
 {
+	if (!cChunkDef::IsValidHeight(a_BlockPos.y))
+	{
+		return false;
```

Comment:
I have a slight concern here that we might end up with use of uninitialised memory as a consequence of this return - idiomatic use of GBTM does not initialise the arguments, and with an early return they are not initialised here either. 

I think we should zero them here for safety.

Reply:
I could be wrong, but I think there are other cases where early return does not initialize the arguments; for example, if `!Chunk->IsValid()`. I don't entirely understand when a Chunk would be invalid, but that's another case where the arguments will be left uninitialized. Also, most (all?) C++ compilers should zero the values anyway afaik, since they're enums, although I'm not totally sure.

Reply:
You're right, we need to add an initialisation to the invalid chunk case as well. Most C++ compilers do zero-initialise variables, but that is non-standard undefined behaviour which we should not rely on.

Reply:
I think the modus operandi "If the return value is `false`, then the out params are uninitialized" is valid. It should just probably be mentioned in the doxy-comment. All the callsites should be doing `if (!GetBlockTypeMeta(...)) { return ...; }` so the uninitialized values don't matter.

Reply:
Perhaps, is there some marker that would tell the compilers that ignoring the return value of this function is a bad thing?